File: benchmarks/solutions/go/a1-two-sum.go
package main

func twoSum(nums []int, target int) (int, int) {
	seen := make(map[int]int)
	for i, n := range nums {
		complement := target - n
		if j, ok := seen[complement]; ok {
			return j, i
		}
		seen[n] = i
	}
	return -1, -1
}

File: benchmarks/solutions/zig/a1-two-sum.zig
const std = @import("std");

fn twoSum(nums: []const i32, target: i32) [2]usize {
    var seen = std.AutoHashMap(i32, usize).init(std.heap.page_allocator);
    defer seen.deinit();

    for (nums, 0..) |n, i| {
        const complement = target - n;
        if (seen.get(complement)) |j| {
            return .{ j, i };
        }
        seen.put(n, i) catch unreachable;
    }
    unreachable;
}
